libs/api/src/lib/graphql/subscription-public.ts

```diff
@@ -51,6 +52,39 @@ export const GraphQLPublicSubscription = new GraphQLObjectType<SubscriptionWithR
       resolve: createProxyingResolver(async (subscription, _, {urlAdapter}) => {
         return await urlAdapter.getSubscriptionURL(subscription)
       })
+    },
+    canExtend: {
+      type: new GraphQLNonNull(GraphQLBoolean),
+      resolve: createProxyingResolver(async (subscription, _, context) => {
+        const [paymentMethod, unpaidInvoice] = await Promise.all([
+          context.loaders.paymentMethodsByID.load(subscription.paymentMethodID),
+          context.prisma.invoice.findFirst({
+            where: {
+              subscription: {
+                userID: subscription.userID
+              },
+              OR: [{canceledAt: null}, {paidAt: null}]
+            }
+          })
+        ])
+
+        /**
+         * Can only extend when:
+         *   Subscription is extendable
+         *   Subscription is not deactivated
+         *   Subscription is about to run out (less than 1 month)
+         *   All invoices have been paid (or cancelled)
+         *   Not using a deprecated payment method
+         */
+        return (
+          subscription.extendable &&
+          !subscription.deactivation &&
+          +add(new Date(), {months: 1}) > +subscription.paidUntil &&
```

Comment:
Logic doesn't work as expected. Either in front-end or here. Please test.

Tested 7th of January 2025 for the following screenshot. And also tested with paidUntil 5th of January and 5th of March. None of them works.
![image](https://github.com/user-attachments/assets/f3b35025-c384-4cd8-8ec6-98ec936b6cf2)

Reply:
The logic was actually correct, but the query above for the unpaid invoice wasn't